scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/CloneRepositoryImpl.java

```diff
@@ -0,0 +1,129 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import java.io.File;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+
+import javax.transaction.Transactional;
+
+import org.springframework.stereotype.Service;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.InvalidDataAccessResourceUsageException;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import edu.asu.diging.scriptoocloud.core.data.ClonedRepository;
+import edu.asu.diging.scriptoocloud.core.forms.CloneForm;
+import edu.asu.diging.scriptoocloud.core.model.Repository;
+import edu.asu.diging.scriptoocloud.core.model.impl.RepositoryImpl;
+import edu.asu.diging.scriptoocloud.core.service.CloneRepository;
+import edu.asu.diging.simpleusers.core.model.IUser;
+
+@Transactional
+@Service
+public class CloneRepositoryImpl implements CloneRepository{
+
+    @Autowired 
+    private ClonedRepository clonedRepository;
+    
+    private String host;
+    private String owner;
+    private String repo;
+    private String requester;
+    private String url;
+    private ZonedDateTime creationDate;
+    private String drive = "C:/";
+    
+    @Override
+    public ArrayList<Repository> getRepos(){
+        Iterable<RepositoryImpl> repoModels = clonedRepository.findAll();
+        ArrayList<Repository> reposList = new ArrayList<>();
+        repoModels.iterator().forEachRemaining(r -> reposList.add(r));
+        return reposList;
+    }
+
+    @Override
+    public String cloneRepo(CloneForm cloneForm){    
+        
+        host = cloneForm.getHost();
+        owner = cloneForm.getOwner();
+        repo = cloneForm.getRepo();
+        requester = ((IUser)SecurityContextHolder.getContext()
```

Comment:
I would pass the user information as a parameter to decouple the business logic from spring security